src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1195,6 +1223,100 @@ def construct_payload(self):
         super().construct_payload()
         self.set_up_service_binds()
         self.set_up_extended_location()
+        self.set_up_source()
+        self.set_up_repo()
+
+    def set_up_source(self):
+        if self.get_argument_source():
+            self._set_up_source_or_repo()
+
+    def set_up_repo(self):
+        if self.get_argument_repo():
+            # Check if container app exists and if it does, set the image to the image of the existing container app
+            container_app = self._get_containerapp_if_exists()
+            if container_app:
+                # Executing the create command with --repo makes two ARM calls. The container app is created first
+                # and the source control is set only after the container app is created. This means, when the container app is
+                # first created, the image would be defaulted to mcr.microsoft.com/k8se/quickstart:latest (if --image was not provided)
+                # and the image name would change to `registry-server/containerapp-name:{github-sha}` only
+                # after the source control completes successfully
+                # and the new image is deployed.
+                # If the create command is executed multiple times however, we need to make sure that the image is not defaulted
+                # to mcr.microsoft.com/k8se/quickstart:latest if source control fails the subsequent times by setting the image
+                # to the previously deployed image.
```

Comment:
Is there a test case for checking executed multiple times the image is not defaulted to `mcr.microsoft.com/k8se/quickstart:latest`?

Reply:
Unfortunately, we do not have tests for source control in the CLI because there is a dependency on GitHub to authenticate with the token generated by GitHub before setting up source control and requires manual intervention.
It is currently not possible to mock this. This is why we do not have e2e tests for `--repo` while testing `up` either.